### PR TITLE
Added Course creation feature

### LINKED ISSUE #42: Task.16 - Create a button to open course creation page on the teacher home page
Create a button component for the teacher's home page that, when clicked, navigates to the course creation page.

### LINKED ISSUE #43: Task.17 - Create a course creation page where course details can be specified
Create a page to create new course where the teacher can enter the following details:
 - Course id
 - Course number
 - Department
 
This page can be accessed by clicking the "Create a Course" button #42 .
The route #44 will allow the new course to be added to the database.

### LINKED ISSUE #44: Task.19 - Create express route to create a new course for a teacher
- Develop a secure Express route to handle new course creation
- Ensure that the course ID is unique by checking against existing courses
- Validate successful course creation

### PR DESCRIPTION
**Pull request by srabm**
closes #43

### CODE REVIEW

File: client/src/App.jsx [hunk 1]
```diff
@@ -10,6 +10,10 @@ import StudentLogin from "./pages/StudentLogin";
 import TeacherLogin from "./pages/TeacherLogin";
 import "./App.css";
 
+{/*--------TEMPORARY-----*/}
+import CourseCreation from "./pages/CourseCreation";
+{/*--------TEMPORARY-----*/}
+
```
Comment (Marian-Hristov) [inline client/src/App.jsx]: remember to remove this!

File: client/src/App.jsx [hunk 2]
```diff
@@ -23,6 +27,11 @@ function App() {
           <Route path="/teacherreg" element={<TeacherRegistration />} />
           <Route path="/studentlogin" element={<StudentLogin />} />
           <Route path="/teacherlogin" element={<TeacherLogin />} />
+
+          {/*--------TEMPORARY-----*/}
+          <Route path="/course-creation" element={<CourseCreation />} />
```
Comment (Marian-Hristov) [inline client/src/App.jsx]: remember that this is supposed to be a component that is going to be nested within another component and not another page!

**Comment by Vinuyans** [comment]
Will Merge #45 and  #58 into this branch so all issues related to creation page can be all together before merging. This will let us debug easier as well.

**Review by Marian-Hristov — COMMENTED** [review]
Backend looks good to me, @Vinuyans can you review the front end?

**Review by Vinuyans — APPROVED** [review]
lgtm

**Pull request merged**